packages/react/src/combobox/root/ComboboxRootContext.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import type { FloatingRootContext } from '../../floating-ui-react';
+import type { BaseOpenChangeReason } from '../../utils/translateOpenChangeReason';
+import { ComboboxStore } from '../store';
+import { HTMLProps } from '../../utils/types';
+import type { useFieldControlValidation } from '../../field/control/useFieldControlValidation';
+
+export type ValueChangeReason = 'item-press' | 'input-change';
+
+export interface ComboboxRootContext {
+  select: 'single' | 'multiple' | 'none';
+  mounted: boolean;
+  selectedValue: any;
+  setSelectedValue: (
+    value: any,
+    event: Event | undefined,
+    reason: ValueChangeReason | undefined,
+  ) => void;
+  open: boolean;
+  setOpen: (
+    open: boolean,
+    event: Event | undefined,
+    reason: BaseOpenChangeReason | undefined,
+  ) => void;
```

Comment:
You should aim to have a single point of truth for state. `open` and `selectedValue` should come from the store, those are unstable values that change the context, so they force a re-render for all its consumers. I also see more state values below that could be moved to the store.

The context should ideally contain only stable values like refs or setters, otherwise you'll end up with re-renders and the whole `Store` logic is wasted.